projects/gigya/src/auth/facade/gigya-auth.service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import {
+  AuthService,
+  StateWithAuth,
+  AuthActions,
+  OCC_USER_ID_CURRENT,
+} from '@spartacus/core';
+import { GigyaAuthActions } from '../store/actions';
+import { take } from 'rxjs/operators';
+
+declare var gigya: any;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GigyaAuthService extends AuthService {
+  constructor(protected store: Store<StateWithAuth>) {
+    super(store);
+  }
+
+  /**
+   * Loads a new user token using custom oauth flow
+   *
+   * @param UID
+   * @param UIDSignature
+   * @param signatureTimestamp
+   * @param idToken
+   * @param baseSite
+   */
+  authorizeWithCustomGigyaFlow(
+    UID: string,
+    UIDSignature: string,
+    signatureTimestamp: string,
+    idToken: string,
+    baseSite: string
+  ): void {
+    this.store.dispatch(
+      new GigyaAuthActions.LoadGigyaUserToken({
+        UID: UID,
+        UIDSignature: UIDSignature,
+        signatureTimestamp: signatureTimestamp,
+        idToken: idToken,
+        baseSite: baseSite,
+      })
+    );
+  }
+
+  /**
+   * Logout a storefront customer
+   */
+  logout(): void {
+    this.getUserToken()
```

Comment:
Instead of copying you can use `super.logout()`

Reply:
Made the required change to call super implementation.